.github/workflows/build-plugin-archive.yml

```
@@ -0,0 +1,183 @@
+        uses: actions/github-script@v6
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: wp-cli
+
+      - name: Install Composer dependencies with dev dependencies
+        # ⬑ The idea is to ensure we have non-production tools like composer-assets-compiler and translation downloader available
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: --prefer-dist
+
+      - name: Check optional Composer build tools
+        id: composer-tools
+        run: |
+          hasDep(){
+            composer show -i -N -D --strict "${1}" >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          echo "assets-compiler=$( hasDep inpsyde/composer-assets-compiler )" >> $GITHUB_OUTPUT
+          echo "translation-downloader=$( hasDep inpsyde/wp-translation-downloader )" >> $GITHUB_OUTPUT
+
+      - name: Run Composer Assets Compiler
+        if: steps.composer-tools.outputs.assets-compiler == '0'
+        run: composer compile-assets
+
+      - name: Run WP Translation Downloader
+        if: steps.composer-tools.outputs.translation-downloader == '0'
+        run: composer wp-translation-downloader:download
+
+      - name: Install Composer dependencies without dev dependencies
+        # ⬑ Now we want to get rid of dev dependencies and most probably skip composer scripts
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+      - name: Check for package.json
+        id: check-for-package
+        run: |
+          if [ -f package.json ]; then
+            echo "has_package=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_package=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up node
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: ${{ inputs.PACKAGE_MANAGER }}
+
+      - name: Add commit hash to plugin header
+        env:
+          GIT_SHA: ${{ steps.git.outputs.sha_short }}
+        run: 'sed -Ei "s/SHA: .*/SHA: ${GIT_SHA}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      - name: Set plugin version header
+        env:
+            PLUGIN_VERSION: ${{ inputs.PLUGIN_VERSION }}
+        run: 'sed -Ei "s/Version: .*/Version: ${PLUGIN_VERSION}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+      - name: Install dist-archive command
+        run: wp package install wp-cli/dist-archive-command
+
+      - name: Execute custom code before archive creation
+        run: |
+            ${{ inputs.EXTRA_PHP_FILE }}
+
+      - name: Run wp-cli command
```

The official name is "WP-CLI".
```suggestion
      - name: Run WP-CLI command
```